src/main/kotlin/lotto/controller/LottoController.kt

```diff
@@ -1,47 +1,44 @@
 package lotto.controller
 
 import lotto.adapter.LottoInputAdapter
-import lotto.domain.LottoBallMachine
 import lotto.domain.LottoGame
 import lotto.domain.LottoPurchaseAmount
+import lotto.domain.ProfitRateCalculator
 import lotto.response.LottoLinesResponse
 import lotto.response.LottoRankResponse
-import lotto.view.InputView
 import lotto.view.OutputView
 
 class LottoController(
-    private val inputView: InputView,
+    private val inputAdapter: LottoInputAdapter,
     private val outputView: OutputView,
-    lottoBallMachine: LottoBallMachine,
+    private val profitRateCalculator: ProfitRateCalculator,
 ) {
-    private val adapter = LottoInputAdapter()
-    private val lottoGame: LottoGame
-    private val lottoPurchaseAmount: LottoPurchaseAmount
-
-    init {
-        val purchaseInput = inputView.inputPurchaseAmount()
-        lottoPurchaseAmount = adapter.adaptPurchaseAmount(purchaseInput)
-        lottoGame = LottoGame(lottoPurchaseAmount, lottoBallMachine)
+    fun getLottoPurchaseAmount(): LottoPurchaseAmount {
+        return inputAdapter.fetchPurchaseAmount()
     }
 
-    fun announcePurchasedLotto() {
+    fun announcePurchasedLotto(lottoGame: LottoGame) {
         outputView.printPurchaseCount(lottoGame.getPurchaseCount())
         val lottoLines = lottoGame.getLottoLines()
         val lottoLinesResponse = LottoLinesResponse(lottoLines)
         outputView.printPurchaseLottoLines(lottoLinesResponse)
     }
 
-    fun announceGameResult() {
-        val winningLineInput = inputView.inputWinningNumbers()
-        val winningNumbers = adapter.adaptWinningNumbers(winningLineInput)
-        val gameResult = lottoGame.returnGameResult(winningNumbers)
-        val lottoProfitRate = gameResult.makeLottoProfitRate(lottoPurchaseAmount.toLottoPurchaseCount())
+    fun announceGameResult(
+        lottoGame: LottoGame,
+        lottoPurchaseAmount: LottoPurchaseAmount,
+    ) {
+        val winningBalls = inputAdapter.fetchWinningNumbers()
+        val bonusBall = inputAdapter.fetchBonusNumber()
+        val gameResult = lottoGame.returnGameResult(winningBalls, bonusBall, profitRateCalculator)
+        val profitRate = gameResult.calculateProfitRate(lottoPurchaseAmount)
 
         val gameResultResponse =
             gameResult.extractResult().map {
                 LottoRankResponse(it.first, it.second)
             }
```

Comment:
이 로직도 좀 더 OOP 패러다임에 맡게 내부 값 추출 및 연산작업을 캡슐화 해본다면 다음과 같이 작성해 볼 수도 있을 것 같네요.
```suggestion
        val gameResultResponse =
            gameResult.mapResult {
                LottoRankResponse(it.first, it.second)
            }
```

Reply:
이렇게 되면 domain인 LottoGameRseult가 LottoRankResponse를 반환하게 되는데, 
dto와 domain 간의 의존성 방향에 대해서는 어떻게 생각하시나요?



+LottoRankResponse에서 LottoGameResult를 받아 List<LottoRankResponse>를 생성하는 방식도 있을 것 같은데,
전 이것도 어색하게 느껴저서 dto의 컬렉션 클래스 LottoRankResponses 만들어 위와 같은 로직을 위치시키는 편입니다.
한솔님은 특정 클래스가 자신의 컬렉션 클래스를 반환하는 함수를 가지고 있는 것에 대해서 어떻게 생각하시나요?

Reply:
종화님 해당 방식은, GameResult가 DTO를 의존하는 방식이 아닙니다. 
해당 행위는 Stream API의 map과 동일한 것으로 GameResult라는 도메인에서 자신의 특정 값을 추상화된 행위 람다식을 매개변수로 받아 변환 후 반환해준다는 행위로 볼 수 있습니다. 

저희가 List에서 Stream API의 map 종단 오퍼레이터를 사용하면서 dto변환을 한다고 List나 Stream에서 dto를 의존한다고 하지 않는것처럼요

Reply:
```
fun <T> extractResult(transform: (Pair<LottoRank, Int>) -> T): List<T> {
        return getLottoRankCounts()
            .toList()
            .sortedByDescending { it.first.rank }
            .map(transform)
    }
```

결국 LottoRankResponse 대신 제네릭을 통한 추상화를 말씀하시는 걸까요?